### PR TITLE
Unify events into one handler

### PR DESCRIPTION
**Pull request by dcvz**
you will now register one handler (like Android)

`TrackPlayer.registerEventHandler((event) => {})`

which will return all the different events defined in the wiki :)

### CODE REVIEW

File: lib/index.js [hunk 1]
```diff
@@ -48,6 +48,36 @@ function remove(tracks) {
     return TrackPlayer.remove(tracks);
 }
 
+function registerEventHandler(handler) {
+    if (Platform.OS !== 'android') {
+        function warpResponse(event, payload) {
+            // transform into headlessTask format and return to handler
+            const additionalKeys = payload || {};
+            handler({ type: event, ...additionalKeys });
```
Comment (Guichaguri) [inline lib/index.js]: "handler" is a function that returns the event handler function. It's looks unnecessary, but that's how react native works

```js
AppRegistry.registerHeadlessTask('TrackPlayer', () => {
    return async (data) {
        // Handle event
    }
});
```

I guess it's to only initialize the event handler code when it needs to.
https://facebook.github.io/react-native/docs/headless-js-android.html#the-js-api

File: lib/index.js [hunk 2]
```diff
@@ -48,6 +48,36 @@ function remove(tracks) {
...
+
+            'remote-play',
+            'remote-next',
+            'remote-stop',
+            'remote-pause',
+            'remote-previous',
+        ]
+
+        for (let i = 0; i < events.length; i++) {
+            eventEmitter.addListener(event, (payload) => { warpResponse(event, payload) });
```
Comment (Guichaguri) [inline lib/index.js]: `event` does not exist?

I think you can replace `(payload) => { warpResponse(event, payload) }` with `wrapResponse.bind(null, events[i])`

**Comment by dcvz** [comment]
want to take up another look @Guichaguri

File: lib/index.js [hunk 3]
```diff
@@ -71,10 +71,10 @@ function registerEventHandler(handler) {
         ]
 
         for (let i = 0; i < events.length; i++) {
-            eventEmitter.addListener(event, (payload) => { warpResponse(event, payload) });
+            eventEmitter.addListener(warpResponse.bind(null, events[i]));
```
Comment (Guichaguri) [inline lib/index.js]: You still need the event name as the first parameter in `addListener`

**Pull request merged**